RNWCPP/ReactUWP/Views/ReactControl.cpp

```diff
@@ -101,9 +110,13 @@ void ReactControl::AttachRoot() noexcept
   // Register callback from instance for live reload
   m_liveReloadCallbackCookie = m_reactInstance->RegisterLiveReloadCallback([this]()
   {
-    m_reactInstance->DefaultNativeMessageQueueThread()->runOnQueue([this]()
+    auto weakThis = m_weakThis.This();
```

Comment:
If we're going to have this new shareptr wrapper thing in a mainline path, should reactcontrol be used as a sharedptr instead of unique_ptr?  then we can shared_from_this instead?

Reply:
no. It doesn't matter if ReactControl is shared or unique_ptr.
I played tricky to workaround the problem: the actual shared data is 'This' other than ReactControl.
At the beginning, This point to ReactControl, When ReactControl is destroyed, ThisHelper is destroyed first, and This is pointing to null. But This may still exist, and it point to null.

We have this layout:
ReactRootView
 ReactControl
   ThisHelper 
        shared_ptr<This>
            This -> this, and null if ThisHelper is destroyed.

From the implementation, the actual weak_ptr is 'This' other than this(or ReactControl), but customer can think it's the same, because I provide StrongThis function to deduce 'this' from This.

I was tried to make shared_ptr<ReactControl>, it makes me that I need to ReactControl knows about ReactRootView because ReactControl doesn't have the ability to make a weak_ptr.